samples/angular-todo/src/app/features/journey/form/form.component.ts

```diff
@@ -79,14 +94,35 @@ export class FormComponent implements OnInit {
     this.submittingForm = true;
 
     try {
+      const selectIdCallback = step?.callbacks.find(
+        (cb) => cb.payload.type === 'SelectIdPCallback',
+      ) as SelectIdPCallback;
+
+      if (selectIdCallback) {
+        const nameCallBacksInputs = step?.payload.callbacks.find(
+          (cb) => cb.type === 'NameCallback',
+        )?.input;
+        const idToken2Input = nameCallBacksInputs?.find((input) => input.name === 'IDToken2').value;
+
+        if (this.isIdentityProviderLogin(selectIdCallback) && idToken2Input !== '') {
+          selectIdCallback.payload.input[0].value = 'localAuthentication';
+        }
+      }
```

Comment:
I'd really like to see some code comments in here helping the reader step through the logic of what we are trying to achieve here. It took me a bit to realize that this was using certain callback values to determine if we need to set `localAuthentication` as the value for the Identity Provider if they have filled in the name input.

Reply:
of course not a problem. This has to be added because when pressing enter it selects the first identity provider by default so it was either disable the default behavior for submitting or keep the option up to submit the form when pressing enter.

Reply:
done